### PR TITLE
feat(zk_toolbox): Add zk_supervisor config-writer command

### PR DESCRIPTION
**Pull request by matias-gonz**
## What ❔
Add zk_supervisor override configs command

## Why ❔
In some cases we want to override some default configs. Right now it is done manually and we need proper automation for regular use/ci.

### CODE REVIEW

File: zk_toolbox/crates/common/src/yaml.rs
Comment (Deniallugo) [inline zk_toolbox/crates/common/src/yaml.rs:1]: it's a confusing name, better to rename it to `config writer` or something simillar

File: zk_toolbox/crates/common/src/yaml.rs [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+            key3: null
+            "#,
+        )
+        .unwrap();
+
+        let expected: serde_yaml::Value = serde_yaml::from_str(
+            r#"
+            key1: value1
+            key2: value2
+            key3: null
```
Comment (Deniallugo) [inline zk_toolbox/crates/common/src/yaml.rs:457]: can we remove this flag instead?

**Comment by Deniallugo** [comment]
As an integration test you can remove this sed 
https://github.com/matter-labs/zksync-era/blob/3609ea6f1b6d767c0f1d64a9303f0331db920931/.github/workflows/ci-core-reusable.yml#L129-L131

File: etc/env/file_based/overrides/legacy-new.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+db:
+  merkle_tree:
+    mode: LIGHTWEIGHT
+experimental_vm:
+  state_keeper_fast_vm_mode: NEW
+mempool:
+  delay_interval: 50
```
Comment (perekopskiy) [inline etc/env/file_based/overrides/legacy-new.yaml:7]: nit: config itself has nothing to do with legacy, I would change name to loadtest-new.yaml and loadtest-old.yaml

File: etc/env/file_based/overrides/no_proofs.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+eth:
+  sender:
+    proof_sending_mode: ONLY_REAL_PROOFS
```
Comment (perekopskiy) [inline etc/env/file_based/overrides/no_proofs.yaml:3]: ONLY_REAL_PROOFS and no_proofs are completely different

File: etc/env/file_based/overrides/legacy-new.yaml
Comment (Deniallugo) [inline etc/env/file_based/overrides/legacy-new.yaml:1]: let's at least put it under tests/folder

File: etc/env/file_based/overrides/validium.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+eth:
+  sender:
+    pubdata_sending_mode: CUSTOM
+state_keeper_config:
+  pubdata_overhead_part: 0.0
```
[the remark this answers is not in the document]
Comment (matias-gonz) [reply etc/env/file_based/overrides/validium.yaml:5]: Looks like we missed that, it is added now

**Review by Deniallugo — APPROVED** [review]

**Review by perekopskiy — APPROVED** [review]

**Review by Deniallugo — APPROVED** [review]

**Review by artmakh — APPROVED** [review]

**Pull request merged**